sdk/azidentity/azure_cli_credential.go

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+type AzureCLITokenProvider interface {
+	GetCLIToken(ctx context.Context, resource string) ([]byte, error)
+}
+
+type azureCLITokenProvider struct{}
```

Comment:
If this empty type is just to provide a single method, you can implement the method as a first-class function and remove this type.

Reply:
Done